Cleanup warnings

```
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/EthDroid.java:10: error: package org.ethereum.geth does not exist
import org.ethereum.geth.Account;
                        ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/EthDroid.java:11: error: package org.ethereum.geth does not exist
import org.ethereum.geth.Context;
                        ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/EthDroid.java:12: error: package org.ethereum.geth does not exist
import org.ethereum.geth.EthereumClient;
                        ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/EthDroid.java:13: error: package org.ethereum.geth does not exist
import org.ethereum.geth.Geth;
                        ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/EthDroid.java:14: error: package org.ethereum.geth does not exist
import org.ethereum.geth.Header;
                        ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/EthDroid.java:15: error: package org.ethereum.geth does not exist
import org.ethereum.geth.Node;
                        ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/EthDroid.java:16: error: package org.ethereum.geth does not exist
import org.ethereum.geth.SyncProgress;
                        ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/EthDroid.java:18: error: package rx does not exist
import rx.Observable;
         ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/EthDroid.java:27: error: cannot find symbol
    private Context mainContext;
            ^
  symbol:   class Context
  location: class EthDroid
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/EthDroid.java:28: error: cannot find symbol
    private Node node;
            ^
  symbol:   class Node
  location: class EthDroid
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/ChainConfig.java:3: error: package org.ethereum.geth does not exist
import org.ethereum.geth.Enodes;
                        ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/ChainConfig.java:4: error: package org.ethereum.geth does not exist
import org.ethereum.geth.Geth;
                        ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/ChainConfig.java:5: error: package org.ethereum.geth does not exist
import org.ethereum.geth.NodeConfig;
                        ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/EthDroid.java:30: error: cannot find symbol
    private EthereumClient client;
            ^
  symbol:   class EthereumClient
  location: class EthDroid
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/KeyManager.java:5: error: package org.ethereum.geth does not exist
import org.ethereum.geth.Account;
                        ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/KeyManager.java:6: error: package org.ethereum.geth does not exist
import org.ethereum.geth.Accounts;
                        ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/KeyManager.java:7: error: package org.ethereum.geth does not exist
import org.ethereum.geth.Geth;
                        ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/KeyManager.java:8: error: package org.ethereum.geth does not exist
import org.ethereum.geth.KeyStore;
                        ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/EthDroid.java:32: error: cannot find symbol
    private Account mainAccount;
            ^
  symbol:   class Account
  location: class EthDroid
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/Transaction.java:7: error: package org.ethereum.geth does not exist
import org.ethereum.geth.Account;
                        ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/Transaction.java:8: error: package org.ethereum.geth does not exist
import org.ethereum.geth.Address;
                        ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/Transaction.java:9: error: package org.ethereum.geth does not exist
import org.ethereum.geth.BigInt;
                        ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/Transaction.java:10: error: package org.ethereum.geth does not exist
import org.ethereum.geth.Block;
                        ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/Transaction.java:11: error: package org.ethereum.geth does not exist
import org.ethereum.geth.CallMsg;
                        ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/Transaction.java:12: error: package org.ethereum.geth does not exist
import org.ethereum.geth.Context;
                        ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/Transaction.java:13: error: package org.ethereum.geth does not exist
import org.ethereum.geth.Geth;
                        ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/Transaction.java:14: error: package org.ethereum.geth does not exist
import org.ethereum.geth.Hash;
                        ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/Transaction.java:15: error: package org.ethereum.geth does not exist
import org.ethereum.geth.KeyStore;
                        ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/Transaction.java:17: error: package okio does not exist
import okio.ByteString;
           ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/Transaction.java:18: error: package rx does not exist
import rx.Observable;
         ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/EthDroid.java:56: error: cannot find symbol
    public Observable<Header> newHeadFilter(){
           ^
  symbol:   class Observable
  location: class EthDroid
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/EthDroid.java:56: error: cannot find symbol
    public Observable<Header> newHeadFilter(){
                      ^
  symbol:   class Header
  location: class EthDroid
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/EthDroid.java:64: error: cannot find symbol
    public Balance getBalanceOf(Account account) throws Exception {
                                ^
  symbol:   class Account
  location: class EthDroid
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/EthDroid.java:167: error: cannot find symbol
    public Context getMainContext() {
           ^
  symbol:   class Context
  location: class EthDroid
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/EthDroid.java:170: error: cannot find symbol
    public EthereumClient getClient() {
           ^
  symbol:   class EthereumClient
  location: class EthDroid
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/EthDroid.java:176: error: cannot find symbol
    public Account getMainAccount() {
           ^
  symbol:   class Account
  location: class EthDroid
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/EthDroid.java:189: error: cannot find symbol
    private static void setMainAccount(EthDroid eth,Account mainAccount){
                                                    ^
  symbol:   class Account
  location: class EthDroid
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/EthDroid.java:205: error: cannot find symbol
    public void setMainAccount(Account mainAccount) {
                               ^
  symbol:   class Account
  location: class EthDroid
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/ChainConfig.java:28: error: cannot find symbol
    NodeConfig nodeConfig;
    ^
  symbol:   class NodeConfig
  location: class ChainConfig
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/KeyManager.java:20: error: cannot find symbol
    private KeyStore keystore;
            ^
  symbol:   class KeyStore
  location: class KeyManager
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/KeyManager.java:33: error: cannot find symbol
    public Account newAccount(String passphrase) throws Exception {
           ^
  symbol:   class Account
  location: class KeyManager
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/KeyManager.java:37: error: cannot find symbol
    public Account newUnlockedAccount(String passphrase) throws Exception{
           ^
  symbol:   class Account
  location: class KeyManager
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/KeyManager.java:43: error: cannot find symbol
    public List<Account> getAccounts() throws Exception{
                ^
  symbol:   class Account
  location: class KeyManager
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/KeyManager.java:52: error: cannot find symbol
    public boolean accountExists(Account account){
                                 ^
  symbol:   class Account
  location: class KeyManager
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/KeyManager.java:57: error: cannot find symbol
    public boolean accountIsLocked(Account account){
                                   ^
  symbol:   class Account
  location: class KeyManager
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/KeyManager.java:67: error: cannot find symbol
    public void lockAccount(Account account) throws Exception {
                            ^
  symbol:   class Account
  location: class KeyManager
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/KeyManager.java:71: error: cannot find symbol
    public void unlockAccount(Account account,String passphrase) throws Exception{
                              ^
  symbol:   class Account
  location: class KeyManager
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/KeyManager.java:75: error: cannot find symbol
    public void unlockAccountDuring(Account account,String passphrase,long seconds) throws Exception {
                                    ^
  symbol:   class Account
  location: class KeyManager
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/KeyManager.java:79: error: cannot find symbol
    public void deleteAccount(Account account,String passphrase) throws Exception{
                              ^
  symbol:   class Account
  location: class KeyManager
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/KeyManager.java:83: error: cannot find symbol
    public void updateAccountPassphrase(Account account,String passphrase,String newPassphrase) throws Exception{
                                        ^
  symbol:   class Account
  location: class KeyManager
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/KeyManager.java:87: error: cannot find symbol
    public byte[] signString(Account account,String toSign) throws Exception{
                             ^
  symbol:   class Account
  location: class KeyManager
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/KeyManager.java:92: error: cannot find symbol
    public byte[] unlockAndsignString(Account account,String password,String toSign) throws Exception{
                                      ^
  symbol:   class Account
  location: class KeyManager
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/KeyManager.java:97: error: cannot find symbol
    public KeyStore getKeystore() {
           ^
  symbol:   class KeyStore
  location: class KeyManager
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/Transaction.java:38: error: cannot find symbol
    private Address to;
            ^
  symbol:   class Address
  location: class Transaction
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/Transaction.java:39: error: cannot find symbol
    private Account from;
            ^
  symbol:   class Account
  location: class Transaction
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/Transaction.java:41: error: cannot find symbol
    private BigInt value;
            ^
  symbol:   class BigInt
  location: class Transaction
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/Transaction.java:42: error: cannot find symbol
    private BigInt gas;
            ^
  symbol:   class BigInt
  location: class Transaction
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/Transaction.java:43: error: cannot find symbol
    private BigInt gasPrice;
            ^
  symbol:   class BigInt
  location: class Transaction
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/Transaction.java:46: error: cannot find symbol
    private Context txContext;
            ^
  symbol:   class Context
  location: class Transaction
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/Transaction.java:78: error: cannot find symbol
    public Transaction to(Address account){
                          ^
  symbol:   class Address
  location: class Transaction
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/Transaction.java:86: error: cannot find symbol
    public Transaction from(Account account, String passphrase){
                            ^
  symbol:   class Account
  location: class Transaction
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/Transaction.java:92: error: cannot find symbol
    public Transaction from(Account account){
                            ^
  symbol:   class Account
  location: class Transaction
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/Transaction.java:98: error: cannot find symbol
    public Transaction value(BigInt value){
                             ^
  symbol:   class BigInt
  location: class Transaction
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/Transaction.java:105: error: cannot find symbol
    public Transaction gasAmount(BigInt gas){
                                 ^
  symbol:   class BigInt
  location: class Transaction
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/Transaction.java:109: error: cannot find symbol
    public Transaction gasPrice(BigInt gasPrice){
                                ^
  symbol:   class BigInt
  location: class Transaction
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/Transaction.java:129: error: cannot find symbol
    public Transaction context(Context context){
                               ^
  symbol:   class Context
  location: class Transaction
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/Transaction.java:145: error: package org.ethereum.geth does not exist
    public org.ethereum.geth.Transaction getRawTransaction() throws Exception {
                            ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/Transaction.java:150: error: package org.ethereum.geth does not exist
    private org.ethereum.geth.Transaction sign() throws Exception{
                             ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/Transaction.java:161: error: cannot find symbol
    private CallMsg toCallMessage() throws Exception{
            ^
  symbol:   class CallMsg
  location: class Transaction
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/Transaction.java:178: error: cannot find symbol
    public Hash send() throws Exception{
           ^
  symbol:   class Hash
  location: class Transaction
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/Transaction.java:184: error: cannot find symbol
    public Observable<Block> sendWithNotification() throws Exception{
           ^
  symbol:   class Observable
  location: class Transaction
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/Transaction.java:184: error: cannot find symbol
    public Observable<Block> sendWithNotification() throws Exception{
                      ^
  symbol:   class Block
  location: class Transaction
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/EthDroid.java:148: error: cannot find symbol
        public Builder withMainAccount(Account mainAccount){
                                       ^
  symbol:   class Account
  location: class Builder
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/ChainConfig.java:221: error: cannot find symbol
        private static Enodes listToEnodes(List<String> enodesList){
                       ^
  symbol:   class Enodes
  location: class Builder
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/Utils.java:3: error: package org.ethereum.geth does not exist
import org.ethereum.geth.Hash;
                        ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/Utils.java:4: error: package org.ethereum.geth does not exist
import org.ethereum.geth.Transactions;
                        ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/Utils.java:193: error: cannot find symbol
    public static boolean transactionListContains(Transactions transactions, Hash txHash) throws Exception {
                                                  ^
  symbol:   class Transactions
  location: class Utils
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/Utils.java:193: error: cannot find symbol
    public static boolean transactionListContains(Transactions transactions, Hash txHash) throws Exception {
                                                                             ^
  symbol:   class Hash
  location: class Utils
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/solidity/element/event/SolidityEvent.java:18: error: package okio does not exist
import okio.ByteString;
           ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/solidity/element/event/SolidityEvent.java:19: error: package rx does not exist
import rx.Observable;
         ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/FilterOptions.java:3: error: package org.ethereum.geth does not exist
import org.ethereum.geth.Address;
                        ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/FilterOptions.java:4: error: package org.ethereum.geth does not exist
import org.ethereum.geth.Addresses;
                        ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/FilterOptions.java:5: error: package org.ethereum.geth does not exist
import org.ethereum.geth.FilterQuery;
                        ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/FilterOptions.java:6: error: package org.ethereum.geth does not exist
import org.ethereum.geth.Geth;
                        ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/FilterOptions.java:7: error: package org.ethereum.geth does not exist
import org.ethereum.geth.Hashes;
                        ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/FilterOptions.java:8: error: package org.ethereum.geth does not exist
import org.ethereum.geth.Topics;
                        ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/solidity/element/event/SolidityEvent.java:44: error: cannot find symbol
    Observable<SType[]> createFilterAndDecode() throws Exception {
    ^
  symbol:   class Observable
  location: class SolidityEvent<T>
  where T is a type-variable:
    T extends SType declared in class SolidityEvent
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/solidity/element/event/SolidityEvent.java:55: error: cannot find symbol
    public Observable<SingleReturn<T>> listen() throws Exception {
           ^
  symbol:   class Observable
  location: class SolidityEvent<T>
  where T is a type-variable:
    T extends SType declared in class SolidityEvent
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/FilterOptions.java:17: error: cannot find symbol
    FilterQuery query;
    ^
  symbol:   class FilterQuery
  location: class FilterOptions
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/FilterOptions.java:32: error: cannot find symbol
    public FilterOptions addAddress(Address address){
                                    ^
  symbol:   class Address
  location: class FilterOptions
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/FilterOptions.java:53: error: cannot find symbol
    public FilterQuery getQuery() {
           ^
  symbol:   class FilterQuery
  location: class FilterOptions
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/model/FilterOptions.java:57: error: cannot find symbol
    private static Hashes fromListToHashes(List<String> list) throws Exception {
                   ^
  symbol:   class Hashes
  location: class FilterOptions
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/solidity/element/function/SolidityFunction.java:15: error: package org.ethereum.geth does not exist
import org.ethereum.geth.Block;
                        ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/solidity/element/function/SolidityFunction.java:16: error: package org.ethereum.geth does not exist
import org.ethereum.geth.Hash;
                        ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/solidity/element/function/SolidityFunction.java:25: error: package rx does not exist
import rx.Observable;
         ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/solidity/element/function/SolidityFunction.java:72: error: cannot find symbol
    public Hash send() throws Exception {
           ^
  symbol:   class Hash
  location: class SolidityFunction<T>
  where T is a type-variable:
    T extends SType declared in class SolidityFunction
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/solidity/element/function/SolidityFunction.java:75: error: cannot find symbol
    public Observable<Block> sendWithNotification() throws Exception {
           ^
  symbol:   class Observable
  location: class SolidityFunction<T>
  where T is a type-variable:
    T extends SType declared in class SolidityFunction
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/solidity/element/function/SolidityFunction.java:75: error: cannot find symbol
    public Observable<Block> sendWithNotification() throws Exception {
                      ^
  symbol:   class Block
  location: class SolidityFunction<T>
  where T is a type-variable:
    T extends SType declared in class SolidityFunction
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/solidity/SolidityUtils.java:4: error: package com.google.gson does not exist
import com.google.gson.JsonArray;
                      ^
/ethdroid/ethdroid/src/main/java/io/ethmobile/ethdroid/solidity/SolidityUtils.java:5: error: package com.google.gson does not exist
import com.google.gson.JsonObject;
100 warnings
```

--- a/ethdroid/build.gradle
+++ b/ethdroid/build.gradle
@@ -35,24 +35,34 @@ android {
 
 dependencies {
     ext.gethDep = 'org.ethereum:geth:1.6.3'
+    ext.googleDepsVersion = '25.1.0'
 
+    // For compilation
     compile fileTree(include: ['*.jar'], dir: 'libs')
-    androidTestCompile('com.android.support.test.espresso:espresso-core:2.2.2', {
-        exclude group: 'com.android.support', module: 'support-annotations'
-    })
-    androidTestCompile 'com.android.support:support-annotations:25.1.0'
-    androidTestCompile 'com.android.support.test:runner:0.5'
-    androidTestCompile 'com.android.support.test:rules:0.5'
-    compile 'com.android.support:appcompat-v7:25.1.0'
-    testCompile 'junit:junit:4.12'
-    compileOnly gethDep
-    androidTestCompile gethDep
-    javadocDeps gethDep
     compile 'io.reactivex:rxjava:1.2.4'
     compile 'io.reactivex:rxandroid:1.2.1'
     compile 'com.google.code.gson:gson:2.8.0'
     compile 'com.squareup.okio:okio:1.13.0' /* ByteString.decodeHex */
+    compile "com.android.support:appcompat-v7:$googleDepsVersion"
 
+    // For compilation, but provided at runtime
+    // Configure also them for javadoc generation bellow
+    compileOnly gethDep
+
+    // For javadoc generation
+    javadocDeps gethDep
+
+    // For tests only
+    testCompile 'junit:junit:4.12'
+
+    // For integration tests only
+    androidTestCompile('com.android.support.test.espresso:espresso-core:2.2.2', {
+        exclude group: 'com.android.support', module: 'support-annotations'
+    })
+    androidTestCompile "com.android.support:support-annotations:$googleDepsVersion"
+    androidTestCompile 'com.android.support.test:runner:0.5'
+    androidTestCompile 'com.android.support.test:rules:0.5'
+    androidTestCompile gethDep
 }
 
 if (project.hasProperty("android")) { // Android libraries

--- a/ethdroid/build.gradle
+++ b/ethdroid/build.gradle
@@ -1,3 +1,8 @@
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.io.FileOutputStream
+import java.util.zip.ZipFile
+
 apply plugin: 'com.android.library'
 apply plugin: 'com.github.dcendents.android-maven'
 
@@ -76,6 +81,25 @@ if (project.hasProperty("android")) { // Android libraries
         classpath += project.files(android.getBootClasspath().join(File.pathSeparator))
         classpath += configurations.compile
         classpath += configurations.javadocDeps
+        afterEvaluate {
+            // Wait after evaluation to add the android classpath
+            // to avoid "buildToolsVersion is not specified" error
+            classpath += files(android.getBootClasspath())
+
+            // Process AAR dependencies
+            def aarDependencies = classpath.filter { it.name.endsWith('.aar') }
+            classpath -= aarDependencies
+            aarDependencies.each { aar ->
+                // Extract classes.jar from the AAR dependency, and add it to the javadoc classpath
+                def outputPath = "$buildDir/tmp/aarJar/${aar.name.replace('.aar', '.jar')}"
+                classpath += files(outputPath)
+
+                // Use a task so the actual extraction only happens before the javadoc task is run
+                dependsOn task(name: "extract ${aar.name}").doLast {
+                    extractEntry(aar, 'classes.jar', outputPath)
+                }
+            }
+        }
     }
 } else { // Java libraries
     task sourcesJar(type: Jar, dependsOn: classes) {
@@ -87,6 +111,25 @@ if (project.hasProperty("android")) { // Android libraries
 task javadocJar(type: Jar, dependsOn: javadoc) {
     classifier = 'javadoc'
     from javadoc.destinationDir
+}
+
+// Utility method to extract only one entry in a zip file
+private def extractEntry(archive, entryPath, outputPath) {
+    if (!archive.exists()) {
+        throw new GradleException("archive $archive not found")
+    }
+
+    def path = Paths.get(outputPath)
+    if (!Files.exists(path)) {
+        def zip = new ZipFile(archive)
+        zip.entries().each {
+            if (it.name == entryPath) {
+                Files.createDirectories(path.getParent())
+                Files.copy(zip.getInputStream(it), path)
+            }
+        }
+        zip.close()
+    }
 }
 
 artifacts {

--- a/ethdroid/build.gradle
+++ b/ethdroid/build.gradle
@@ -1,6 +1,5 @@
 import java.nio.file.Files
 import java.nio.file.Paths
-import java.io.FileOutputStream
 import java.util.zip.ZipFile
 
 apply plugin: 'com.android.library'

--- a/ethdroid/build.gradle
+++ b/ethdroid/build.gradle
@@ -5,6 +5,10 @@ android {
     compileSdkVersion 25
     buildToolsVersion "25.0.2"
 
+    configurations {
+        javadocDeps
+    }
+
     defaultConfig {
         minSdkVersion 21
         targetSdkVersion 25
@@ -30,6 +34,8 @@ android {
 }
 
 dependencies {
+    ext.gethDep = 'org.ethereum:geth:1.6.3'
+
     compile fileTree(include: ['*.jar'], dir: 'libs')
     androidTestCompile('com.android.support.test.espresso:espresso-core:2.2.2', {
         exclude group: 'com.android.support', module: 'support-annotations'
@@ -39,8 +45,9 @@ dependencies {
     androidTestCompile 'com.android.support.test:rules:0.5'
     compile 'com.android.support:appcompat-v7:25.1.0'
     testCompile 'junit:junit:4.12'
-    compileOnly 'org.ethereum:geth:1.6.3'
-    androidTestCompile 'org.ethereum:geth:1.6.3'
+    compileOnly gethDep
+    androidTestCompile gethDep
+    javadocDeps gethDep
     compile 'io.reactivex:rxjava:1.2.4'
     compile 'io.reactivex:rxandroid:1.2.1'
     compile 'com.google.code.gson:gson:2.8.0'
@@ -58,6 +65,7 @@ if (project.hasProperty("android")) { // Android libraries
         source = android.sourceSets.main.java.srcDirs
         classpath += project.files(android.getBootClasspath().join(File.pathSeparator))
         classpath += configurations.compile
+        classpath += configurations.javadocDeps
     }
 } else { // Java libraries
     task sourcesJar(type: Jar, dependsOn: classes) {

--- a/ethdroid/build.gradle
+++ b/ethdroid/build.gradle
@@ -57,6 +57,7 @@ if (project.hasProperty("android")) { // Android libraries
     task javadoc(type: Javadoc) {
         source = android.sourceSets.main.java.srcDirs
         classpath += project.files(android.getBootClasspath().join(File.pathSeparator))
+        classpath += configurations.compile
     }
 } else { // Java libraries
     task sourcesJar(type: Jar, dependsOn: classes) {

--- a/ethdroid/src/main/java/io/ethmobile/ethdroid/ChainConfig.java
+++ b/ethdroid/src/main/java/io/ethmobile/ethdroid/ChainConfig.java
@@ -104,7 +104,7 @@ public class ChainConfig {
         /**
          * Set the size of the blockchain database in cache
          * Cf. //TODO
-         * @param dbCache
+         * @param dbCache Megabytes of memory allocated to internal caching
          * @return reference on the parametrized builder
          */
         public Builder withDbCache(long dbCache){

--- a/ethdroid/src/main/java/io/ethmobile/ethdroid/EthDroid.java
+++ b/ethdroid/src/main/java/io/ethmobile/ethdroid/EthDroid.java
@@ -70,8 +70,11 @@ public class EthDroid {
         EthDroid build;
 
         /**
-         * Parameterized Builder with the default values :
-         * - Context : @withDefaultContext
+         * Parametrized Builder with the default values :
+         * <ul>
+         *     <li>Context : @{@link Builder#withCancelContext()}</li>
+         * </ul>
+         * @param datadir Directory where to store Geth files
          */
         public Builder(String datadir) {
             build = new EthDroid();

--- a/ethdroid/src/main/java/io/ethmobile/ethdroid/model/Balance.java
+++ b/ethdroid/src/main/java/io/ethmobile/ethdroid/model/Balance.java
@@ -23,8 +23,8 @@ public class Balance {
 
     /**
      * Build a balance object with the given value (in wei === 10^18 ether)
-     * @param value
-     * @return
+     * @param value Value ti use in wei (1 wei === 10^18 ether)
+     * @return balance built from value
      */
     public static Balance of(long value){
         return new Balance(value);

--- a/ethdroid/src/main/java/io/ethmobile/ethdroid/model/Transaction.java
+++ b/ethdroid/src/main/java/io/ethmobile/ethdroid/model/Transaction.java
@@ -118,14 +118,20 @@ public class Transaction {
     /**
      * Set transaction data from given hexadecimal byte array in string format
      * data can be prefixed by '0x' hexadecimal identifier
-     * @param data
-     * @return
+     * @param data String date used to build the transaction
+     * @return Transaction built from data
      */
     public Transaction data(String data){
         if( data.contains("0x") ) data = data.substring(2);
         data(ByteString.decodeHex(data).toByteArray());
         return this;
     }
+
+    /**
+     * Set context in the transaction
+     * @param context Context to set in the transaction
+     * @return Updated current transaction
+     */
     public Transaction context(Context context){
         this.txContext = context;
         return this;
@@ -141,6 +147,7 @@ public class Transaction {
      * Get raw geth transaction.
      * Returned transaction is not signed, so it can't be sent.
      * @return not signed transaction
+     * @throws Exception An error occured
      */
     public org.ethereum.geth.Transaction getRawTransaction() throws Exception {
         if( to == null ) throw new EthDroidException(NO_RECIPIENT_ERROR);

